hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -138,6 +173,182 @@ def hpat_pandas_series_iloc_impl(self):
     return hpat_pandas_series_iloc_impl
 
 
+@overload_method(SeriesType, 'nsmallest')
+def hpat_pandas_series_nsmallest(self, n=5, keep='first'):
+    """
+    Pandas Series method :meth:`pandas.Series.nsmallest` implementation.
+
+    .. only:: developer
+       Test: python -m hpat.runtests -k hpat.tests.test_series.TestSeries.test_series_nsmallest*
+
+    n: :obj:`int`, default 5
+        Return this many ascending sorted values.
+    keep: :obj:`str`, default 'first'
+        When there are duplicate values that cannot all fit in a Series of n elements:
+        first : return the first n occurrences in order of appearance.
+        last : return the last n occurrences in reverse order of appearance.
+        all : keep all occurrences. This can result in a Series of size larger than n.
+        *unsupported*
+
+    Returns
+    -------
+    :obj:`series`
+         returns :obj:`series`
+    """
+
+    _func_name = 'Method nsmallest().'
+
+    ty_checker = TypingChecker(_func_name)
+    ty_checker.check_series(self)
+    ty_checker.check_integer(n, 'n')
+    ty_checker.check_string(keep, 'keep')
+
+    dtype = self.data.dtype
+    if isinstance(self.index, types.NoneType):
+        inner_list_type = types.ListType(types.intp)
+        def hpat_pandas_series_nsmallest_impl(self, n=5, keep='first'):
+            if keep != 'first':
+                raise ValueError("Method nsmallest(). Unsupported parameter. "
+                                 "Given 'keep' != 'first'")
+
+            nsmallest = hpat.hiframes.api.nlargest(self._data, n, False,
+                                                   hpat.hiframes.series_kernels.lt_f)
+            indices = Dict.empty(dtype, inner_list_type)
+            for idx, item in enumerate(self._data):
+                if item not in indices:
+                    indices[item] = List.empty_list(types.intp)
+                indices[item].append(idx)
+
+            max_length = len(nsmallest)
+            local_index = []
+            for item in numpy.unique(nsmallest):
+                for index in indices[item]:
+                    if len(local_index) == max_length:
+                        break
+                    local_index.append(index)
+
+            return pandas.Series(nsmallest, local_index, name=self._name)
```

Comment:
I'm afraid you changing name of the type, not label the Series here.
To set a name to the Series object you have to implement parameter `name` in Series::ctor(). Not sure it will work or not because base class `types.IterableType` has the same name of a parameter. In this case it might be required to change field name to `label`.